### PR TITLE
Update createRef to match the new React 16.3 api

### PR DESCRIPTION
**Pull request by Markionium**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Since the api for `React.createRef` changed from the initial PR/discussion on which we based the polyfill our version was out of sync with the recently announced version in React 16.3.

This PR takes care of that and updates the `createRef` function in utilities to match the React one.

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by Markionium** [comment]
It's fairly sure it's not going to change again as it's now released as part of the latest React version.

I think it's ok to add a deprecation and continue to support `value` for a while. The fact that you gave workshops on it definitely makes that point very valid. :)

**Comment by Markionium** [comment]
@micahgodbolt I've added a `.value` prop that can be used as a getter for `.current`. :) (When we decide it's been enough time to migrate we can remove it)

File: packages/utilities/src/createRef.ts [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
+ * This is a polyfill for the React.createRef() api.
+ * For more info on React.createRef() see the official React documentation
+ * on creating and accessing refs.
+ * @see https://reactjs.org/docs/refs-and-the-dom.html#creating-refs
+ * @see https://reactjs.org/docs/refs-and-the-dom.html#accessing-refs
+ */
 export function createRef<T>(): RefObject<T> {
   const refObject = ((element: T | null): void => {
-    refObject.value = element;
+    refObject.current = element;
```
Comment (dzearing) [inline packages/utilities/src/createRef.ts]: Can we keep backwards compatibility, and populate value as well?

**Comment by dzearing** [comment]
Thank you for updating it!!

**Pull request merged**